### DISCUSSION TITLE
blog/serialize-response-nestjs

### DISCUSSION THREAD

**Discussion by giscus** [adarshaacharya/portfolio-giscus#4] [Announcements]
# blog/serialize-response-nestjs

Learn how to serialize Nest.js API response using class-transformer and class-validator

https://www.adarshaacharya.com.np/blog/serialize-response-nestjs

**Comment by mukezhz** [comment]
Nice article it was really insightful.

Thank you for the article.

**Reply by adarshaacharya** [reply]
thanks for feedback !

**Comment by toviazs** [comment]
Hi! I really enjoyed the explanation, it was very clear. I was using this approach until I encountered an issue. What should I do if the same endpoint needs to return different responses based on a parameter? For example, a /users endpoint that, when requested by an admin, returns more detailed information compared to when it's requested by a regular user.

**Reply by adarshaacharya** [reply]
in that case, it will be better to not exclude anything and return based on the role of user.

**Reply by toviazs** [reply]
Thank you! ✨🚀
